int16[12] q_matrix     # Q matrix
#uint16[3]  i_principal_rate  # principal of inertia
int16[2]  rotate_angle # cos_theta, sin_theta

